model/user/user.go

```diff
@@ -166,6 +168,53 @@ func (u *DBUser) UpdateParsleySettings(settings parsley.Settings) error {
 	return nil
 }
 
+// CheckAndUpdateSchedulingLimit checks if the number of tasks to be activated by the user is allowed given
+// the global per-user hourly task scheduling limit, and updates relevant timestamp and counter info used
+// to track the user's hourly scheduling usage. The numTasksActivated parameter being negative signifies
```

Comment:
do you mean numTasksModified? (allowing this to be negative seems confusing, would it be complex to split this into two parameters?)

Reply:
Yes, missed that in the refactor. I added an additional `activated` boolean to try and make it a bit clearer.